external-crates/move/crates/move-package-alt/src/git/cache.rs

```diff
@@ -753,4 +775,41 @@ pkg_git = {{ git = "{}", rev = "{short_sha}" }}
                 .is_ok()
         )
     }
+
+    #[tokio::test]
+    async fn test_tag_checkout() {
+        let tag = "releases/1";
+
+        let project = git::new("git_repo", |project| {
+            project.file("packages/pkg_a/Move.toml", &basic_manifest("a", "0.0.1"))
+        });
+
+        project.add_tag(tag);
+
+        let cache_dir = tempdir().unwrap();
+        let cache = GitCache::new_from_dir(cache_dir.path());
+
+        let git_tree = cache
+            .resolve_to_tree(project.root_path_str(), &Some(tag.to_string()), None)
+            .await
+            .unwrap();
+
+        let result = git_tree.fetch().await;
+        assert!(result.is_ok());
+
+        let another_tag = "releases/2";
+        project.add_tag(another_tag);
+
+        let git_tree = cache
+            .resolve_to_tree(
+                project.root_path_str(),
+                &Some(another_tag.to_string()),
+                None,
+            )
+            .await
+            .unwrap();
+
+        let result = git_tree.fetch().await;
+        assert!(result.is_ok());
+    }
```

Comment:
Apropos to the previous comment, you may want to add a test where there is a branch named `whetever/{rev}`, which should not get checked out.